bitnami/opensearch/values.yaml

```diff
@@ -3408,3 +3408,181 @@ dashboards:
+    ## E.g.
+    ## selector:
+    ##   matchLabels:
+    ##     app: my-app
+    ##
+    selector: {}
+    ## @param snapshots.persistence.size Size of data volume
+    ##
+    size: 8Gi
+    ## @param snapshots.persistence.storageClass Storage class of backing PVC
+    ## If defined, storageClassName: <storageClass>
+    ## If set to "-", storageClassName: "", which disables dynamic provisioning
+    ## If undefined (the default) or set to null, no storageClassName spec is
+    ##   set, choosing the default provisioner.  (gp2 on AWS, standard on
+    ##   GKE, AWS & OpenStack)
+    ##
+    storageClass: ""
+    ## @param snapshots.persistence.subPath The subdirectory of the volume to mount to, useful in dev environments and one PV for multiple services
+    ##
+    subPath: ""
+  ## Configure Pods Security Context
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod
+  ## @param snapshots.podSecurityContext.enabled Enabled data pods' Security Context
+  ## @param snapshots.podSecurityContext.fsGroupChangePolicy Set filesystem group change policy
+  ## @param snapshots.podSecurityContext.sysctls Set kernel settings using the sysctl interface
+  ## @param snapshots.podSecurityContext.supplementalGroups Set filesystem extra groups
+  ## @param snapshots.podSecurityContext.fsGroup Set snapshots pod's Security Context fsGroup
+  ##
+  podSecurityContext:
+    enabled: true
+    fsGroupChangePolicy: Always
+    sysctls: []
+    supplementalGroups: []
+    fsGroup: 1001
+  ## @param snapshots.policies [default: SNAPSHOT_POLICY_API_BODY] Each object represents a snapshot policy in YAML form, which will be converted to JSON and then passed as the HTTP body data to the OpenSearch REST API.
+  ## NOTE: The field `snapshot_config.repository` in each policy must match a repo name in `snapshots.repositories`.
+  ##
+  policies:
+    default:
+      creation:
+        schedule:
+          cron:
+            expression: "0 20 * * *"
+            timezone: UTC
+      deletion:
+        condition:
+          max_age: 7d
+          min_count: 1
+        schedule:
+          cron:
+            expression: 0 20 * * *
+            timezone: UTC
+      description: Default snapshot policy
+      enabled: true
+      snapshot_config:
+        repository: default # This must match a repo name in `snapshots.repositories`
+  ## @param snapshots.repositories [default: SNAPSHOT_REPO_API_BODY] Each object represents a snapshot repository in YAML form, which will be converted to JSON and then passed as the HTTP body data to the OpenSearch REST API.
+  ## NOTE: The field `settings.location` in each repo must match the snapshot repo path configured in `snapshotRepoPath`, if its `type` is `fs`.
+  ##
+  repositories:
+    default:
+      settings:
+        location: /snapshots # This must match the snapshot repo path configured in `snapshotRepoPath`
+      type: fs
```

Comment:
I'd like some input on this design decision. Is it good to have the API body object as YAML like this? On the one hand, I like that everything is YAML so it's easy to work with. On the other hand, this kind of hides the fact that these are not "regular" values but actually YAML representations of OpenSearch API payloads. Might be clearer if we include them as JSON strings, but I think that would make it easier for users to introduce typos in their configs.

WDYT?

Reply:
As this is a yaml file, I think this configuration format is correct but you can use any other format here and change the template as required.

Reply:
Certainly. I was just wondering what would provide the best user experience for the end user/consumer of the chart.

It also makes a difference for how we document the parameters. I think it's bad if the `@param` for this object becomes a copy-paste of the OpenSearch documentation. That will be a nightmare for you to maintain. If we just document this as a opaque blob, it's easier to say "this object must be an API object according to the OpenSearch documentation" and then they can look up the examples there.

I'll try and make it a JSON implementation to see if that looks better. I'll get back once I've updated my subjective opinion on the matter.

Reply:
I just realized that my sanity checks (in [policies-cm.yaml](https://github.com/bitnami/charts/blob/c29ade4c05217605be43471a1338d1352c68af2c/bitnami/opensearch/templates/snapshots/policies-cm.yaml#L20) and [repos-cm.yaml](https://github.com/bitnami/charts/blob/c29ade4c05217605be43471a1338d1352c68af2c/bitnami/opensearch/templates/snapshots/repos-cm.yaml#L19-L24)) are entirely dependent on parsing the YAML. I think they are valuable to have, and it would be very hard to implement them unless we keep these objects as YAML code.

Thus, I'm marking this as resolved. If you disagree, please unresolve!